source/computer/editor.rst

```diff
@@ -0,0 +1,215 @@
+编辑器
+=======
+
+:本节贡献者: |姚家园|\（作者）、
+             |田冬冬|\（审稿）
+:最近更新日期: 2021-02-13
+:预计花费时间: 30 分钟
+
+----
+
+我们可以在终端输入命令交给系统执行，但要编辑文本或者编写代码，则需要使用一个叫做编辑器的程序，
+即一个允许移动光标在屏幕上编辑文字的程序。编辑器是科研人员写代码和和管理系统配置文件的重要工具。
+
+学习编辑器
+----------
+
+日常科研中，我们大部分时间都花在编写代码和普通文件上，所以花点时间掌握某个适合自己的编辑器是
+非常值得的。通常学习和掌握一个新的编辑器的步骤如下：
+
+- 阅读教程：例如，官方文档、本教程等
+- 坚持使用它来完成所有的编辑工作：一般而言，刚开始会让工作效率降低
+- 随时查阅：如果某个操作看起来像是有更方便的实现方法，一般情况下真的会有
+
+如果遵循上述步骤，并且坚持使用新的编辑器完成所有的编辑任务，则学习一个复杂的编辑器的过程
+一般是这样的：
+
+- 前两个小时学习编辑器的基本操作。例如，打开和编辑文件、保存与退出等
+- 学习时间累计达到一定时间后（如 20 个小时），使用新编辑器的效率应该已经和使用老编辑器一样快
+- 此后，新编辑器地益处开始显现，因为有了足够的知识和肌肉记忆后，使用新编辑器将大大节省时间
+  
+现代编辑器都是一个复杂且强大的工具，永远有新东西可学：学的越多，效率越高。
+
+选择编辑器
+----------
+
+该学哪个编辑器？这个问题没有标准答案，选择其中一个一直用下去，遇到需要重复劳动或者不顺心的地方
+就去找各种配置和插件。笔者在这里只是根据自身经验，做一点总结和推荐。
+
+编辑器按照操作方式可以分为两种类型：基于命令行的和图形化的。例如，`vim <https://www.vim.org/>`__
+是一款基于命令行的较为流行的编辑器，`Visual Studio Code <https://code.visualstudio.com/>`__\
+（VS Code）是一款图形化的较为流行的编辑器。
+
+编辑器按照功能可以又大致分为三种类型：文本编辑器、代码编辑器以及集成开发环境（Integrated Development Environment，简称 IDE）。
+例如，vim 一般算为文本编辑器，VS Code 算为代码编辑器，`PyCharm <https://www.jetbrains.com/pycharm/>`__
+是一款主要用于基于 Python 语言的软件开发的集成开发环境。当然，有时候并不能简单地将某一编辑器归属于其中一种。
+例如，代码编辑器本身也可以很方便地用于编辑文本，代码编辑器也有集成开发环境地部分功能。
+当我们编写代码时，经常在文件间进行切换、阅读、浏览和修改代码，而不是连续编写一大段的文字，
+因此代码编辑器和纯文本编辑器是很不同的两种工具。一般而言，文本编辑器启动速度最快，集成开发环境最慢；
+开发大型软件时，集成开发环境使用起来会更有效率。
+
+Linux 系统一般自带了一个叫做 gedit 的图形化编辑器，但是使用该编辑器来写程序效率太低，不建议使用。
+
+VS Code 是目前最流行的代码编辑器，用于文本编辑时也非常高效，地震学日常科研中推荐使用 VS Code
+编辑文本和代码。
+
+vim 是最流行的基于命令行的编辑器，有时候可能是用户唯一可以使用的编辑器，推荐掌握其基本的编辑功能，
+以便在无法使用 VS Code 或简单文本和代码编辑时使用。
+
+Visual Studio Code
+-------------------
+
+安装和使用
+^^^^^^^^^^
+
+可以参考\ :doc:`setup`\ 中相应操作系统下的“日常软件”安装 VS Code。
+
+刚接触 VS Code 的用户可以参考 VS Code 官方入门教程 `Get Started <https://code.visualstudio.com/docs/getstarted/introvideos>`__
+学习常用用法。使用过程中，可以根据个人习惯配置 VS Code，还可以安装和使用扩展包。
+
+日常科研使用 VS Code 编辑文档和代码遇到疑问时，可以随时查阅，如以下方式：
+
+- 点击菜单栏“帮助”，选择相关选项。例如，“交互式演练场”、“文档”（Visual Studio Code 官方文档）、“键盘快捷键参考”等
+- 使用 Google 搜索
+
+.. dropdown:: :fa:`exclamation-circle,mr-1` 安装 VS Code 中文语言包
+   :container: + shadow
+   :title: bg-info text-white font-weight-bold
+
+   VS Code 的界面默认显示语言是英文，可以安装中文语言包。点击菜单栏“查看”后选择“命令面板”
+   （快捷键：\ :kbd:`Ctrl` + :kbd:`Shift` + :kbd:`p`\ ），接着输入“configure display language”并按
+   :kbd:`Enter` 键，然后选择“安装其他语言”。这时界面会跳转到插件商店自动搜索其他语言，一般第一个就是中文，
+   即“Chinese (Simplified) Language Pack for Visual Studio Code”，点击安装就行了。
+   安装完之后自动重启，界面就变成中文了。
+
+编辑远程计算机文件
```

Comment:
Perhaps we should hide the whole section? Most people don't use servers?

Reply:
Yes~ Done in 8ce65d4.